files/nrpe/check_openstack.py

```diff
@@ -322,10 +343,15 @@ def delete_orphaned_instances(self):
       instance.delete()
 
   def delete_orphaned_floating_ips(self):
-    for tenant_ip in self.nova.floating_ips.list():
-      self.nova.floating_ips.delete(tenant_ip)
-    if len(self.nova.floating_ips.list()) != 0:
-      logging.warn('All floating IPs of instance creation test tenant were not deleted.')
+    # This used to be done with novaclient where it was enought to just list().
+    # With neutronclient one needs to be much more specific, otherwise if ran
+    # with an admin user there is a risk of getting all the Floating IPs.
+    own_project_id = self.session.get_project_id()
+    fip_lookup_params = { 'project_id': own_project_id, }
+    for project_ip in self.neutron.list_floatingips(**fip_lookup_params)['floatingips']:
+      self.neutron.delete_floatingip(project_ip['id'])
+    if len(self.neutron.list_floatingips(**fip_lookup_params)['floatingips']) != 0:
+      logging.warn('All floating IPs of instance creation test project were not deleted.')
```

Comment:
If the value is invalid does it still return all ips and then delete them anyhow.

eg if project_id gets renamed to tenant_id or borris_ip in your search. From a, would a keystone change break this for us.

Reply:
You mean

input: `{ 'project_id': own_project_id, }`
API changed and now expects: `{ 'borris_id': own_project_id, }`

?

That could be a risk. I could see about capping the result set. For example

- Select whichever is smaller, own project Floating IP quota or some reasonably low limit like 5
- If the result set exceeds the selected limit, halt and log an error

Reply:
In the end added a check based on Floating IP ownership.

Reply:
I'm still thinking maybe the admin priv check is safer.